### ISSUE TITLE
mInitialized prevents platform re-init following shutdown

### ISSUE THREAD

**Issue by msandstedt** [project-chip/connectedhomeip#7558]
#### Problem
A flag to prevent platform double-init was introduced with this commit:

```
commit ab8122e4e05f9757a9156c4f23f4a27f715a5bf2
Author: Andrei Litvin <andrei@andy314.com>
Date:   Fri May 7 09:06:11 2021 -0400
    
    * Track platform manager initialization and do not allow double-init
```

This makes absolute sense.  However, `PlatformMgr().Init()`; PlatformMgr().Shutdown(); PlatformMgr().Init()` _should_ be allowed, and currently isn't.

 #### Proposed Solution
Solution is simply to set `mInitialized=false` in PlatformManager.h Shutdown.

Background rationale for continuing to track this in `PlatformManager.h`:

* This is common to all platforms; no platform should be calling Init() twice without first shutting down.
* This cannot be handled in `GenericPlatformManagerImpl` because the flag needs to also prevent platform-specific double-init, and platform-specific is called before `GenericPlatformManagerImpl` Init.

**Comment by msandstedt** [comment]
FYI @andy31415 , @mrjerryjohns ,

The solution should be very simple, and should still prevent platforms from inadvertently double-initing.

**Comment by msandstedt** [comment]
Fixed by #7532.
